blt/src/Blt/Plugin/Commands/Tests/CypressTestCommand.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Cgov\Blt\Plugin\Commands\Tests;
+
+use Acquia\Blt\Robo\Commands\Tests\TestsCommandBase;
+use Acquia\Blt\Robo\Exceptions\BltException;
+use Consolidation\AnnotatedCommand\CommandData;
+use Cgov\CgovNpmTrait;
+
+/**
+ * Defines commands in the "tests" namespace.
+ */
+class CypressTestCommand extends TestsCommandBase {
+  use CgovNpmTrait;
+
+  /**
+   * The standard port number used by Xvfb.
+   *
+   * @var int
+   */
+  private const XVFB_PORT = 99;
+
+  /**
+   * Process ID for the running instance of Xvfb.
+   *
+   * @var int
+   */
+  private $xvfbPid = -9999;
+
+  /**
+   * Base directory for running cypress tests.
+   * This path is relative to the repository root.∂
+   *
+   * @var string
+   */
+  protected $baseDir;
+
+  /**
+   * This hook will fire for all commands in this command file.
+   *
+   * @hook init
+   */
+  public function initialize() {
+    parent::initialize();
+
+    $this->baseDir = $this->getConfigValue("tests.cypress.basePath");
+  }
+
+  /**
+   * Launch point for cypress tests.
+   *
+   * @command tests:cypress
+   * @description Execute all cypress tests.
+   *
+   * @usage -D tests.cypress.paths=${PWD}/tests/cypress/features/
+   *
+   * @aliases cypress tests:cypress:run tcr
+   */
+  public function cypressRunTestsMain() {
+    $commands = [
+      'tests:cypress:npm-install',
+      'tests:cypress:pretest',
+      'tests:cypress:runimpl'
+    ];
+
+    $this->invokeCommands($commands);
+  }
+
+  /**
+   * Set up to run cypress tests
+   *
+   * @command tests:cypress:npm-install
+   * @description Install the NPM packages needed in order to run cypress tests.
+   *
+   */
+  public function cypressSetup() {
+    // If we get tired of the amount of time it takes to do a complete clean install
+    // on every run for local tests, replace this with a call to taskNpmInstall.
+    $result = $this->taskNpmCleanInstall()->dir($this->baseDir)->run();
+    $exit_code = $result->getExitCode();
+
+    if($exit_code) {
+      throw new BltException('Cypress npm install failed.');
+    }
+  }
+
+  /**
+   * Run tests with side-effects the other tests depend on.
+   *
+   * @command tests:cypress:pretest
+   *
+   * @interactGenerateSettingsFiles
+   * @interactInstallDrupal
+   * @validateDrupalIsInstalled
+   * @launchWebServer
+   *
+   * @return void
+   */
+  public function cypressSetuptests() {
```

Comment:
So pre-tests should be run as part of tests. See my comments on the blt.yml

Reply:
Done